clients/src/main/java/org/apache/kafka/common/record/DefaultRecord.java

```diff
@@ -266,7 +272,7 @@ private static long computeChecksum(long timestamp,
 
     @Override
     public boolean hasMagic(byte magic) {
-        return magic >= MAGIC_VALUE_V2;
+        return magic == MAGIC_VALUE_V2;
```

Comment:
I reverted this change for now as the behaviour is specified. I think we should revisit the naming, but let's do it in a separate follow-up.